python/jiminy_py/src/jiminy_py/tree.py

```diff
@@ -173,10 +173,19 @@ def _unflatten_as(data: StructNested[Any],
     """
     data_type = type(data)
     if issubclass_mapping(data_type):  # type: ignore[arg-type]
-        return data_type({  # type: ignore[call-arg]
-            key: _unflatten_as(value, data_leaf_it)
-            for key, value in data.items()  # type: ignore[union-attr]
-        })
+        try:
```

Comment:
I would rather suggest something like this to avoid redundant computations in fallback:
```
flat_items = [(key, _unflatten_as(value, data_leaf_it)) for key, value in data.items()]
try:
    # Initialisation from dict cannot be the default path as `gym.spaces.Dict`
    # would sort keys in this specific scenario, which must be avoided.
    data_type(flat_items)
except (ValueError, RuntimeError):
    # Fallback to initialisation from dict in the rare event of a container 
    # type not supporting initialisation from a sequence of key-value pairs.
    data_type(dict(flat_items))
```

Reply:
Yes it's cleaner